api/src/main/scala/org/ensime/api/config.scala

```diff
@@ -29,60 +28,33 @@ final case class EnsimeConfig(
   /* Proposed alternatives to the legacy wire format field names */
   def root = rootDir
   val referenceSourceJars =
-    (javaSources ++ subprojects.flatMap(_.referenceSourceRoots)).toSet
+    (javaSources ++ projects.flatMap(_.librarySources)).toSet
 
   // some marshalling libs (e.g. spray-json) might not like extra vals
-  val modules = subprojects.map { module => (module.name, module) }.toMap
+  val modules = projects.map { module => (module.id, module) }.toMap
 
   def compileClasspath: Set[File] = modules.values.toSet.flatMap {
-    m: EnsimeModule => m.compileDeps ++ m.testDeps
+    m: EnsimeProject => m.libraryJars
```

Comment:
Will it be okay if we remove these as we bring in code for Analyzer per module? I am just a little hesistant to remove all of it now as they are being used in a few classes and i am not sure about refactoring them.